File: src/main/old/cc/sukazyo/cono/morny/bot/event/OnMedicationNotifyApply.java
package cc.sukazyo.cono.morny.bot.event;

import cc.sukazyo.cono.morny.MornyCoeur;
import cc.sukazyo.cono.morny.bot.api.EventListener;
import cc.sukazyo.cono.morny.daemon.MornyDaemons;
import com.pengrad.telegrambot.model.Message;
import com.pengrad.telegrambot.model.Update;

import javax.annotation.Nonnull;

public class OnMedicationNotifyApply extends EventListener {
	
	@Override
	public boolean onEditedChannelPost (@Nonnull Update update) {
		return editedMessageProcess(update.editedChannelPost());
	}
	
	@Override
	public boolean onEditedMessage (@Nonnull Update update) {
		return editedMessageProcess(update.editedMessage());
	}
	
	private boolean editedMessageProcess (Message edited) {
		if (edited.chat().id() != MornyCoeur.config().medicationNotifyToChat) return false;
		MornyDaemons.medicationTimerInstance.refreshNotificationWrite(edited);
		return true;
	}
	
}

File: src/main/scala/cc/sukazyo/cono/morny/medication_timer/OnMedicationNotifyApply.scala
package cc.sukazyo.cono.morny.medication_timer

import cc.sukazyo.cono.morny.core.MornyCoeur
import cc.sukazyo.cono.morny.core.bot.api.{EventEnv, EventListener}
import com.pengrad.telegrambot.model.Message

class OnMedicationNotifyApply (using instance: MedicationTimer)(using coeur: MornyCoeur) extends EventListener {
	
	override def onEditedMessage (using event: EventEnv): Unit =
		editedMessageProcess(event.update.editedMessage)
	override def onEditedChannelPost (using event: EventEnv): Unit =
		editedMessageProcess(event.update.editedChannelPost)
	
	private def editedMessageProcess (edited: Message)(using event: EventEnv): Unit = {
		if edited.chat.id != coeur.config.medicationNotifyToChat then return;
		if instance.refreshNotificationWrite(edited) then
			event.setEventOk
	}
	
}
